CMake/utility.cmake

```diff
@@ -386,6 +386,51 @@ function(activate_precompiled_headers _PRECOMPILED_HEADER _SOURCE_FILES)
   endif()
 endfunction()
 
+####################################################################################
+# Adds warnings compiler options to the target depending on the category
+# _TARGET Target name
+# _CATEGORY "library" or "unit_tests" or "benchmarks"
+####################################################################################
+macro( set_compiler_warnings _TARGET _CATEGORY)
```

Comment:
There are several things wrong with this function:
1. It is a macro, so you not introducing a new scope, the variables will be available where you call this macro. Prefer functions
2. DRY - e.g. you do this compare: `if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")` 3 times in this macro
3. You repeat yourself also with the warnings (e.g. "/WX" "/W4"), The whole code base should be compiled with the same warning levels.
When there are 3rd Party libs, which does not match this requirement, then we have to address this separately.
But int the current master state, all targets use the same compiler warnings. We can add later more warnings flags.
So in conclusion the category argument can be removed. Keep it simple.
4. Variable naming: `CXX_FLAGS` this are not a generic CXX flags, it should only be used for warnings, so I would call them like this.
5. `CXX_FLAGS_DEBUG` `CXX_FLAGS_RELEASE`, etc. Where are these coming from?